src/ape_plugins/_cli.py

```diff
@@ -22,38 +23,80 @@ def cli():
+        if i.startswith(prefix):
+            start = len(prefix)
+            text = i[start:]
+        else:
+            text = i
+        if version:
+            output.add(f"{text}    ({version})")
+        else:
+            output.add(text)
+    return output
+
+
+@cli.command(name="list", short_help="Display plugins")
 @click.option(
     "-a",
     "--all",
     "display_all",
     default=False,
     is_flag=True,
-    help="Display all plugins (including Core)",
+    help="Display all plugins installed and available (including Core)",
 )
 @ape_cli_context()
 def _list(cli_ctx, display_all):
-    plugins = []
+    installed_second_class_plugins = set()
+    installed_first_class_plugins = set()
+    installed_third_class_plugins = set()
+    version = None
+
     for name, plugin in plugin_manager.list_name_plugin():
-        version_str = ""
         version = get_package_version(name)
 
         if name in FIRST_CLASS_PLUGINS:
             if not display_all:
-                continue  # NOTE: Skip 1st class plugins unless specified
-            version_str = " (core)"
-
-        elif version:
-            version_str = f" ({version})"
-
-        plugins.append(f"{name}{version_str}")
-
-    if plugins:
-        click.echo("Installed plugins:")
-        click.echo("  " + "\n  ".join(plugins))
 
+                continue  # NOTE: Skip 1st class plugins unless specified
+            installed_first_class_plugins.add(f"{name}")
+
+        elif name in SECOND_CLASS_PLUGINS:
+            installed_second_class_plugins.add(f"{name}")
+
+        elif name not in FIRST_CLASS_PLUGINS or name not in SECOND_CLASS_PLUGINS:
+            installed_third_class_plugins.add(f"{name}    ({version})")
+        else:
+            cli_ctx.logger.error(f"{name} is not a plugin.")
+
+    # First Class Plugins
+    if display_all:
+        click.echo("Installed Core Plugins:")
+        installed_first_class_plugins_output = _format_output(
+            installed_first_class_plugins, version=None
+        )
+        click.echo("  {}\n".format("\n  ".join(installed_first_class_plugins_output)))
+
+    # Second Class Plugins
+    available_second = list(SECOND_CLASS_PLUGINS - installed_second_class_plugins)
+
+    if installed_second_class_plugins:
+        installed_second_output = _format_output(installed_second_class_plugins, version)
+        click.echo("(2nd) Installed Second Class Plugins:")
+        click.echo("  {}".format("\n  ".join(installed_second_output)))
     else:
-        cli_ctx.logger.info("No plugins installed")
+        click.echo("No Second Class Plugins installed")
+
+    if display_all:
+        available_second_output = _format_output(available_second, version)
+        if available_second_output:
+            click.echo("\n(2nd) Available Second Class Plugins:")
+            click.echo("  {}".format("\n  ".join(available_second_output)))
+        else:
+            click.echo("You have installed all the Second Class Plugins")
```

Comment:
Hmm I just realized we are not displaying the 3rd class plugins?

Reply:
Ill add it now

Reply:
Was mentioned in a meeting, but they should be included in the `Installed plugins` list.